### PR TITLE
Add relocatable setup.bash script

### LINKED ISSUE #3056: setup.sh script is not relocatable
As noted in https://github.com/osrf/gazebo/issues/2782#issuecomment-656226379 and https://github.com/osrf/gazebo/issues/2782#issuecomment-832710858, the `setup.sh` script installed with gazebo is not relocatable, as it contains install paths [hard-coded at configure time](https://github.com/osrf/gazebo/blob/0fcdfc08fd615e3969c94d4e5c47830eca0d5948/CMakeLists.txt#L166-L189). It looks like @traversaro has already done the hard work of making setup.sh relocatable in https://github.com/robotology/robotology-superbuild-dependencies-vcpkg/blob/master/scripts/setup-gazebo.sh

So we could resolve this ticket by porting that script to this repository.

### PR DESCRIPTION
**Pull request by scpeters**
This refactors the `setup.sh` script to set the `GAZEBO_*_PATH` environment variables relative to a path specified in a `GAZEBO_INSTALL_PREFIX` environment variable (defaulting to the current behavior of setting relative to the configure-time install prefix if `GAZEBO_INSTALL_PREFIX` is empty or unset). It also adds a relocatable `setup.bash` script that uses bash-specific syntax to detect the folder containing the `setup.*sh` scripts, set `GAZEBO_INSTALL_PREFIX` relative to that folder, and then source `setup.sh`.

Fixes #3056.

### CODE REVIEW

File: cmake/setup.sh.in [hunk 1]
```diff
@@ -1,7 +1,22 @@
+# if GAZEBO_INSTALL_PREFIX is unset, fall back to
+# configure-time value of CMAKE_INSTALL_PREFIX
+installPrefix=${GAZEBO_INSTALL_PREFIX:-@CMAKE_INSTALL_PREFIX@}
+if [ ! -d "${installPrefix}" ]; then
+  echo "The install prefix ${installPrefix} does not exist."
+  unset installPrefix
+  return 1
+elif [ ! -f "${installPrefix}/@GAZEBO_RELATIVE_RESOURCE_PATH@/setup.sh" ]; then
```
Comment (jacobperron) [inline cmake/setup.sh.in:8]: Is this checking for the existing of the setup.sh produced by this file? Under what circumstance might this condition be true?

**Review by jacobperron — APPROVED** [review]
LGTM, just one question above.

File: cmake/setup.sh.in (hunk 1 above)
Comment (scpeters) [reply cmake/setup.sh.in:8]: This is checking that the path in the `installPrefix` variable contains at least one expected file. In this case it resolves to `${installPrefix}/share/gazebo-11/setup.sh` for gazebo11, but I suppose we could equivalently choose `${installPrefix}/share/gazebo-11/worlds/empty.world`. The first `if` statement checks that the `$installPrefix` path exists; the second checks for some expected content.

I suppose it's possible to confuse matters if you manually set `GAZEBO_INSTALL_PREFIX` to one install path but source the `setup.sh` from a different install path, but the recommended approach is now to use `setup.bash`, which should avoid that type of confusion

**Review by jacobperron — APPROVED** [review]

**Comment by scpeters** [comment]
CI looks ok to me, merging

**Pull request merged**